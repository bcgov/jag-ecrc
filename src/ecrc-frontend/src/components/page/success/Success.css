.print-page-success {
  cursor: pointer;
  text-decoration: underline;
  color: #1a5a96;
  margin-bottom: 30px;
}

.content-success-sidecard {
  flex-direction: column;
  padding-top: 40px;
  padding-bottom: 30px;
  margin-left: 30px;
}
