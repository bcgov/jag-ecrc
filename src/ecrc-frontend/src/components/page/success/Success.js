/* eslint-disable new-cap */
import React, { useState, useEffect } from "react";
import axios from "axios";
import queryString from "query-string";
import { useLocation, Redirect } from "react-router-dom";
import jsPDF from "jspdf";
import "jspdf-autotable";
import PropTypes from "prop-types";

import Header from "../../base/header/Header";
import Footer from "../../base/footer/Footer";
import Table from "../../composite/table/Table";
import { Button } from "../../base/button/Button";
import {
  isActionPerformed,
  isAuthorized
} from "../../../modules/AuthenticationHelper";

export default function Success({
  page: {
    header,
    applicant: { legalFirstNm, legalSurnameNm },
    org: { orgApplicantRelationship, orgTicketNumber, orgNm },
    applicationInfo: {
      partyId,
      sessionId,
      invoiceId,
      serviceFeeAmount,
      serviceId
    },
    saveApplicationInfo
  }
}) {
  const [toHome, setToHome] = useState(false);
  const location = useLocation();
  const paymentInfo = queryString.parse(location.search);
  const uuid = sessionStorage.getItem("uuid");

  useEffect(() => {
    window.scrollTo(0, 0);

    if (
      !isAuthorized() ||
      !paymentInfo.trnApproved ||
      !isActionPerformed("consent")
    ) {
      setToHome(true);
    }
  }, [paymentInfo.trnApproved]);

  const receiptInfo = [
    { name: "Service Number", value: serviceId },
    { name: "First Name", value: legalFirstNm },
    { name: "Last Name", value: legalSurnameNm },
    { name: "Organization", value: orgNm }
  ];

  if (paymentInfo.trnApproved === "1") {
    receiptInfo.push({ name: "Amount", value: paymentInfo.trnAmount });
  }

  if (orgApplicantRelationship !== "VOLUNTEER") {
    receiptInfo.push({ name: "Date/Time", value: paymentInfo.trnDate });
    receiptInfo.push({
      name: "Transaction ID",
      value: paymentInfo.trnOrderNumber
    });
  }

  const receiptInfoTable = {
    id: "print",
    header: "APPLICATION INFORMATION",
    tableElements: receiptInfo,
    tableStyle: "white"
  };
  // IF PaymentFailure: LogPaumentFailure
  if (paymentInfo.trnApproved === "0") {
    const logFailure = {
      orgTicketNumber,
      requestGuid: uuid,
      serviceId,
      applPartyId: partyId,
      sessionId,
      invoiceId,
      serviceFeeAmount,
      bcepErrorMsg: paymentInfo.messageText
    };

    axios
      .post("/ecrc/private/logPaymentFailure", logFailure)
      .then(() => {})
      .catch(() => {});
  }

  // IF Success and not volunteer: UpdateServiceFinancialTxn?
  // cC_Authorization - Unsure, defer to Shaun
  // payor_Type_Cd - based on application type O for ONETIME, A otherwise
  if (paymentInfo.trnApproved === "1") {
    const token = sessionStorage.getItem("jwt");

    const logSuccess = {
      orgTicketNumber,
      requestGuid: uuid,
      appl_Party_Id: partyId,
      service_Id: serviceId,
      cC_Authorization: paymentInfo.trnId,
      payment_Date: paymentInfo.trnDate,
      payor_Type_Cd: "A",
      payment_Status_Cd: "A",
      session_Id: sessionId,
      invoice_Id: invoiceId,
      transaction_Id: paymentInfo.trnId,
      transaction_Amount: paymentInfo.trnAmount
    };

    axios
      .post("/ecrc/private/updateServiceFinancialTxn", logSuccess, {
        headers: {
          Authorization: `Bearer ${token}`
        }
      })
      .then(() => {})
      .catch(() => {});
  }

  const printButton = {
    label: "Print",
    buttonStyle: "btn ecrc_go_btn",
    buttonSize: "btn btn-sm",
    type: "submit"
  };

  const printAppInfo = () => {
    window.print();
  };

  const pdfButton = {
    label: "Download",
    buttonStyle: "btn ecrc_go_btn",
    buttonSize: "btn btn-sm",
    type: "submit"
  };

  const downloadPDF = () => {
    const doc = new jsPDF();
    doc.autoTable({ theme: "plain", html: "#print" });
    doc.save(`${serviceId}${legalSurnameNm}${legalFirstNm}.pdf`);
  };

  const retryPayment = () => {
    const token = sessionStorage.getItem("jwt");

    axios
      .get(
        `/ecrc/private/getNextInvoiceId?orgTicketNumber=${orgTicketNumber}&requestGuid=${uuid}`,
        {
          headers: {
            Authorization: `Bearer ${token}`
          }
        }
      )
      .then(invoiceResponse => {
        const newInvoiceId = invoiceResponse.data.invoiceId;

        saveApplicationInfo({
          partyId,
          sessionId,
          invoiceId: newInvoiceId,
          serviceFeeAmount,
          serviceId
        });

        const createURL = {
          invoiceNumber: newInvoiceId,
          requestGuid: uuid,
          approvedPage: `${process.env.REACT_APP_FRONTEND_BASE_URL}/ecrc/success`,
          declinedPage: `${process.env.REACT_APP_FRONTEND_BASE_URL}/ecrc/success`,
          errorPage: `${process.env.REACT_APP_FRONTEND_BASE_URL}/ecrc/success`,
          totalItemsAmount: serviceFeeAmount,
          serviceIdRef1: serviceId,
          partyIdRef2: partyId
        };

        return axios.post("/ecrc/private/createPaymentUrl", createURL, {
          headers: { Authorization: `Bearer ${token}` }
        });
      })
      .then(urlResponse => {
        window.location.href = urlResponse.data.paymentUrl;
      });
  };

  if (toHome) {
    return <Redirect to="/" />;
  }

  return (
    <main>
      <Header header={header} />
      <div className="page">
        <div className="content col-md-8">
          <h1>
            {orgApplicantRelationship === "VOLUNTEER" &&
              "Application Submitted"}
            {paymentInfo.trnApproved === "0" && "Payment Declined"}
            {paymentInfo.trnApproved === "1" && "Payment Approved"}
          </h1>
          {paymentInfo.trnApproved !== "0" && (
            <>
              <p>
                Thank you for submitting your application to the Criminal
                Records Review Program.
              </p>
              <p>
                Your application will be reviewed shortly. Once complete, the
                results will be provided directly to the requesting
                organization. We will contact you if further information is
                required.
              </p>
            </>
          )}
          {paymentInfo.trnApproved === "0" && (
            <>
              <span>
                Unfortunately, your payment transaction was declined. Please
                ensure you have the correct credit card information:
              </span>
              <ul className="bodyList">
                <li>16 digit credit card number</li>
                <li>3 digit CVD number</li>
                <li>Non-expired date</li>
                <li>Availale funds to transfer</li>
              </ul>
              <p>
                <button
                  className="notAButton"
                  type="button"
                  onClick={() => retryPayment()}
                >
                  Click here to try again
                </button>
                . Otherwise, please refer to our website for submission options.
              </p>
            </>
          )}
          <div className="print">
            <Table table={receiptInfoTable} />
          </div>
          <Button button={printButton} onClick={printAppInfo} />
          <Button button={pdfButton} onClick={downloadPDF} />
        </div>
        <div className="sidecard" />
      </div>
      <Footer />
    </main>
  );
}

Success.propTypes = {
  page: PropTypes.shape({
    header: PropTypes.shape({
      name: PropTypes.string.isRequired
    }),
    applicant: PropTypes.shape({
      legalFirstNm: PropTypes.string.isRequired,
      legalSurnameNm: PropTypes.string.isRequired
    }),
    org: PropTypes.shape({
      orgApplicantRelationship: PropTypes.string.isRequired,
      orgTicketNumber: PropTypes.string.isRequired,
      orgNm: PropTypes.string.isRequired
    }),
    applicationInfo: PropTypes.shape({
      partyId: PropTypes.number.isRequired,
      sessionId: PropTypes.number.isRequired,
      invoiceId: PropTypes.number.isRequired,
      serviceFeeAmount: PropTypes.number,
      serviceId: PropTypes.number.isRequired
    }),
    saveApplicationInfo: PropTypes.func.isRequired
  }).isRequired
};
