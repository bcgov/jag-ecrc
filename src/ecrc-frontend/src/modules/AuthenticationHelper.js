/* eslint-disable import/prefer-default-export */
import axios from "axios";
import { v4 as uuidv4 } from "uuid";

const jwt = require("jsonwebtoken");

export function isAuthenticated() {
  const token = sessionStorage.getItem("jwt");
  const validator = sessionStorage.getItem("validator");
  let isAuthed = true;

  if (!token || !validator) return false;

  // verify a token symmetric
  jwt.verify(token, validator, err => {
    if (err) {
      isAuthed = false;
    }
  });

  return isAuthed;
}

export function isAuthorized() {
  const token = sessionStorage.getItem("jwt");
  const validator = sessionStorage.getItem("validator");
  let isAuthorize = false;

  if (!token || !validator) return false;

  // verify a token symmetric
  jwt.verify(token, validator, (err, decoded) => {
    if (err || !decoded.authorities) {
      isAuthorize = false;
    } else if (decoded.authorities.includes("Authorized")) {
      isAuthorize = true;
    }
  });

  return isAuthorize;
}

export function isActionPerformed(action) {
  const token = sessionStorage.getItem("jwt");
  const validator = sessionStorage.getItem("validator");
  let isPerformed = false;

  if (!token || !validator) return false;

  // verify a token symmetric
  jwt.verify(token, validator, (err, decoded) => {
    if (err || !decoded.actionsPerformed) {
      isPerformed = false;
    } else if (decoded.actionsPerformed.includes(action)) {
      isPerformed = true;
    }
  });

  return isPerformed;
}

export function storeValidator() {
  const uuid = sessionStorage.getItem("uuid");

  if (!uuid) return false;

  axios
    .get(`/ecrc/initialHandshake?requestGuid=${uuid}`)
    .then(res => {
      const value = res.data;

      if (value) {
        sessionStorage.setItem("validator", value);
      }
    })
    .catch(() => {
      return false;
    });

  return true;
}

export function storeUUID() {
  const uuid = uuidv4();
  sessionStorage.setItem("uuid", uuid);
}

export function generateJWTToken(payload) {
  const validator = sessionStorage.getItem("validator");

  if (!validator) return false;

  let token;

  if (payload.exp) {
    token = jwt.sign(payload, validator);
  } else {
    token = jwt.sign(payload, validator, { expiresIn: "1h" });
  }

  sessionStorage.setItem("jwt", token);

  return token;
}

export function accessJWTToken(token) {
  const validator = sessionStorage.getItem("validator");

  const payload = jwt.verify(token, validator);

  return payload;
}
