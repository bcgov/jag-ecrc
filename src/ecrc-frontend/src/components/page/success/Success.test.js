/* eslint-disable prefer-promise-reject-errors */
import React from "react";
import { create } from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import axios from "axios";
import { render, fireEvent, getByText, wait } from "@testing-library/react";

import Success from "./Success";
import { generateJWTToken } from "../../../modules/AuthenticationHelper";

jest.mock("axios");

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
    listen: jest.fn()
  })
}));

describe("Success Page Component", () => {
  window.scrollTo = jest.fn();

  // Mock browser print
  window.print = jest.fn();
  window.open = jest.fn();

  // Mock window location
  const mockWindow = jest.fn();
  delete window.location;
  window.location = { assign: mockWindow };

  axios.get.mockImplementation(() =>
    Promise.resolve({
      data: {
        invoiceId: "123"
      }
    })
  );

  axios.post.mockImplementation(() =>
    Promise.resolve({
      data: {
        urlResponse: "http://sample.com"
      }
    })
  );

  const header = {
    name: "Criminal Record Check"
  };

  const applicant = {
    legalFirstNm: "Robert",
    legalSurnameNm: "Ross"
  };

  const org = {
    orgApplicantRelationship: "EMPLOYEE",
    orgTicketNumber: "pbs",
    orgNm: "Public Broadcast Network"
  };

  const applicationInfo = {
    partyId: 123,
    sessionId: 456,
    invoiceId: 789,
    serviceFeeAmount: 60,
    serviceId: 987
  };

  const saveApplicationInfo = jest.fn();
  const setError = jest.fn();

  const page = {
    header,
    applicant,
    org,
    applicationInfo,
    saveApplicationInfo,
    setError
  };

  const successUrl =
    "/success?trnApproved=1&trnId=10001213&messageId=1&messageText=Approved&authCode=TEST&responseType=T&trnAmount=28.00&trnDate=3%2F4%2F2020+1%3A01%3A36+PM&trnOrderNumber=2991&trnLanguage=eng&trnCustomerName=&trnEmailAddress=&trnPhoneNumber=&avsProcessed=0&avsId=U&avsResult=0&avsAddrMatch=0&avsPostalMatch=0&avsMessage=Address+information+is+unavailable%2E&cvdId=1&cardType=VI&trnType=P&paymentMethod=CC&ref1=Service+Id%3A+22522&ref2=CRRP+%2D+Org+Party+Id%3A+49119&ref3=&ref4=&ref5=";
  const failureUrl =
    "/success?trnApproved=0&trnId=&messageId=804&messageText=Declined+%2D+Entered+Information+Cannot+Be+Authenticated&authCode=&responseType=T&trnAmount=&trnDate=3%2F2%2F2020+8%3A43%3A52+AM&trnOrderNumber=2946&trnLanguage=eng&trnCustomerName=&trnEmailAddress=&trnPhoneNumber=&avsProcessed=0&avsId=0&avsResult=0&avsAddrMatch=0&avsPostalMatch=0&avsMessage=Address+Verification+not+performed+for+this+transaction%2E&cardType=&trnType=&paymentMethod=&ref1=Service+Id%3A+22478&ref2=CRRP+%2D+Org+Party+Id%3A+49074&ref3=&ref4=&ref5=";

  beforeEach(() => {
    sessionStorage.setItem("validator", "secret");
    sessionStorage.setItem("uuid", "unique123");
    generateJWTToken({
      actionsPerformed: ["consent"],
      authorities: ["Authorized"]
    });
  });

  test("Matches the Volunteer snapshot", () => {
    const successVolunteer = create(
      <MemoryRouter initialEntries={["/success"]}>
        <Success
          page={{
            ...page,
            org: { ...org, orgApplicantRelationship: "VOLUNTEER" }
          }}
        />
      </MemoryRouter>
    );
    expect(successVolunteer.toJSON()).toMatchSnapshot();
  });

  test("Matches the SuccessfulPayment snapshot", () => {
    const successPayment = create(
      <MemoryRouter initialEntries={[successUrl]}>
        <Success page={page} />
      </MemoryRouter>
    );
    expect(successPayment.toJSON()).toMatchSnapshot();
  });

  test("Matches the FailedPayment snapshot", () => {
    const failedPayment = create(
      <MemoryRouter initialEntries={[failureUrl]}>
        <Success page={page} />
      </MemoryRouter>
    );
    expect(failedPayment.toJSON()).toMatchSnapshot();
  });

  test("Validate Retry payment", async () => {
    const { container } = render(
      <MemoryRouter initialEntries={[failureUrl]}>
        <Success page={page} />
      </MemoryRouter>
    );

    fireEvent.click(getByText(container, "Try Again"));

    await wait(() => {
      expect(saveApplicationInfo).toHaveBeenCalled();
    });
  });

  test("Validate Print (on click)", () => {
    const { container } = render(
      <MemoryRouter initialEntries={[successUrl]}>
        <Success page={page} />
      </MemoryRouter>
    );

    fireEvent.click(getByText(container, "Print"));
    expect(window.print).toHaveBeenCalled();
  });

  test("Validate Print (on keydown)", () => {
    const { container } = render(
      <MemoryRouter initialEntries={[successUrl]}>
        <Success page={page} />
      </MemoryRouter>
    );

    fireEvent.keyDown(getByText(container, "Print"));
    expect(window.print).toHaveBeenCalled();
  });

  test("Validate Email", () => {
    const { container } = render(
      <MemoryRouter initialEntries={[successUrl]}>
        <Success page={page} />
      </MemoryRouter>
    );

    fireEvent.click(getByText(container, "Email"));
    expect(window.open).toHaveBeenCalled();
  });

  test("Redirects to error page when not authorized on page load", async () => {
    sessionStorage.clear();

    render(
      <MemoryRouter initialEntries={[failureUrl]}>
        <Success page={page} />
      </MemoryRouter>
    );

    await wait(() => {
      expect(setError).toHaveBeenCalledTimes(1);
    });

    expect(mockHistoryPush).toHaveBeenCalledWith("/criminalrecordcheck/error");
    expect(mockHistoryPush).toHaveBeenCalledTimes(1);
  });

  test("Error case for getNextInvoiceId failing without response data and message", async () => {
    axios.get.mockImplementation(() =>
      Promise.reject({
        response: {
          status: 400
        }
      })
    );

    axios.post.mockImplementation(() =>
      Promise.resolve({
        data: {
          urlResponse: "http://sample.com"
        }
      })
    );

    const { container } = render(
      <MemoryRouter initialEntries={[failureUrl]}>
        <Success page={page} />
      </MemoryRouter>
    );

    fireEvent.click(getByText(container, "Try Again"));

    await wait(() => {
      expect(setError).toHaveBeenCalledTimes(2);
    });

    expect(mockHistoryPush).toHaveBeenCalledWith("/criminalrecordcheck/error");
    expect(mockHistoryPush).toHaveBeenCalledTimes(2);
  });

  test("Error case for getNextInvoiceId failing with response data and message", async () => {
    axios.get.mockImplementation(() =>
      Promise.reject({
        response: {
          status: 400,
          data: {
            message: "This is bad request error"
          }
        }
      })
    );

    axios.post.mockImplementation(() =>
      Promise.resolve({
        data: {
          urlResponse: "http://sample.com"
        }
      })
    );

    const { container } = render(
      <MemoryRouter initialEntries={[failureUrl]}>
        <Success page={page} />
      </MemoryRouter>
    );

    fireEvent.click(getByText(container, "Try Again"));

    await wait(() => {
      expect(setError).toHaveBeenCalledTimes(3);
    });

    expect(mockHistoryPush).toHaveBeenCalledWith("/criminalrecordcheck/error");
    expect(mockHistoryPush).toHaveBeenCalledTimes(3);
  });

  test("Failing logPaymentFailure causes handle error to be called", async () => {
    axios.post.mockImplementation(() =>
      Promise.reject({
        response: {
          status: 400,
          data: {
            message: "This is bad request error"
          }
        }
      })
    );

    render(
      <MemoryRouter initialEntries={[failureUrl]}>
        <Success page={page} />
      </MemoryRouter>
    );

    await wait(() => {
      expect(setError).toHaveBeenCalledTimes(4);
    });

    expect(mockHistoryPush).toHaveBeenCalledWith("/criminalrecordcheck/error");
    expect(mockHistoryPush).toHaveBeenCalledTimes(4);
  });

  test("Failing updateServiceFinancialTxn causes handle error to be called", async () => {
    axios.post.mockImplementation(() =>
      Promise.reject({
        response: {
          status: 400,
          data: {
            message: "This is bad request error"
          }
        }
      })
    );

    render(
      <MemoryRouter initialEntries={[successUrl]}>
        <Success page={page} />
      </MemoryRouter>
    );

    await wait(() => {
      expect(setError).toHaveBeenCalledTimes(5);
    });

    expect(mockHistoryPush).toHaveBeenCalledWith("/criminalrecordcheck/error");
    expect(mockHistoryPush).toHaveBeenCalledTimes(5);
  });

  test("Redirects to host home on cancel click when confirm is selected as Yes", () => {
    const { container } = render(
      <MemoryRouter initialEntries={[failureUrl]}>
        <Success page={page} />
      </MemoryRouter>
    );

    window.confirm = () => true;

    fireEvent.click(getByText(container, "Cancel and Exit"));

    expect(sessionStorage.getItem("jwt")).toBeFalsy();
    expect(mockHistoryPush).toHaveBeenCalledWith("/hosthome");
    expect(mockHistoryPush).toHaveBeenCalledTimes(6);
  });

  test("Does not redirect to host home on cancel click when confirm is selected as No", () => {
    const { container } = render(
      <MemoryRouter initialEntries={[failureUrl]}>
        <Success page={page} />
      </MemoryRouter>
    );

    window.confirm = () => false;

    fireEvent.click(getByText(container, "Cancel and Exit"));

    expect(sessionStorage.getItem("jwt")).toBeTruthy();
  });
});
