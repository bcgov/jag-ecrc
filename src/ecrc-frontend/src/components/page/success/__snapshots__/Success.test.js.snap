// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success Page Component Matches the FailedPayment snapshot 1`] = `
<main>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-none d-md-block"
            height="44"
            src="/images/bcid-logo-rev-en.svg"
            width="181"
          />
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-md-none"
            height="44"
            src="/images/bcid-symbol-rev.svg"
            width="64"
          />
        </a>
        <div>
          <a
            className="navbar-brand nav-item nav-link"
            href="/"
          >
            Criminal Record Check
          </a>
        </div>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNavAltMarkup"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        />
      </div>
    </nav>
  </header>
  <div
    className="page"
  >
    <div
      className="content col-md-8"
    >
      <h1>
        Payment Declined
      </h1>
      <span>
        Unfortunately, your payment transaction was declined. Please ensure you have the correct credit card information:
      </span>
      <ul
        className="bodyList"
      >
        <li>
          16 digit credit card number
        </li>
        <li>
          3 digit CVD number
        </li>
        <li>
          Non-expired date
        </li>
        <li>
          Availale funds to transfer
        </li>
      </ul>
      <p>
        <button
          className="notAButton"
          onClick={[Function]}
          type="button"
        >
          Click here to try again
        </button>
        . Otherwise, please refer to our website for submission options.
      </p>
      <div
        className="print"
      >
        <table
          className="white"
          id="print"
        >
          <thead>
            <tr>
              <th
                colSpan="2"
              >
                APPLICATION INFORMATION
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                Service Number
              </td>
              <td>
                987
              </td>
            </tr>
            <tr>
              <td>
                First Name
              </td>
              <td>
                Robert
              </td>
            </tr>
            <tr>
              <td>
                Last Name
              </td>
              <td>
                Ross
              </td>
            </tr>
            <tr>
              <td>
                Organization
              </td>
              <td>
                Public Broadcast Network
              </td>
            </tr>
            <tr>
              <td>
                Date/Time
              </td>
              <td>
                3/2/2020 8:43:52 AM
              </td>
            </tr>
            <tr>
              <td>
                Transaction ID
              </td>
              <td>
                2946
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <button
        className="btn ecrc_go_btn btn btn-sm"
        onClick={[Function]}
        type="submit"
      >
        Print
        <div
          className="loader-hide"
        >
          <div
            className="btn-loader"
          />
        </div>
      </button>
      <button
        className="btn ecrc_go_btn btn btn-sm"
        onClick={[Function]}
        type="submit"
      >
        Download
        <div
          className="loader-hide"
        >
          <div
            className="btn-loader"
          />
        </div>
      </button>
    </div>
    <div
      className="sidecard"
    />
  </div>
  <footer
    className="footer"
  >
    <nav
      className="navbar navbar-expand-sm navbar-dark"
    >
      <div
        className="container"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
              rel="noopener noreferrer"
              target="_blank"
            >
              Disclaimer
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/privacy"
              rel="noopener noreferrer"
              target="_blank"
            >
              Privacy
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/accessibility"
              rel="noopener noreferrer"
              target="_blank"
            >
              Accessibility
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/copyright"
              rel="noopener noreferrer"
              target="_blank"
            >
              Copyright
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/contact-us"
              rel="noopener noreferrer"
              target="_blank"
            >
              Contact Us
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </footer>
</main>
`;

exports[`Success Page Component Matches the SuccessfulPayment snapshot 1`] = `
<main>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-none d-md-block"
            height="44"
            src="/images/bcid-logo-rev-en.svg"
            width="181"
          />
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-md-none"
            height="44"
            src="/images/bcid-symbol-rev.svg"
            width="64"
          />
        </a>
        <div>
          <a
            className="navbar-brand nav-item nav-link"
            href="/"
          >
            Criminal Record Check
          </a>
        </div>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNavAltMarkup"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        />
      </div>
    </nav>
  </header>
  <div
    className="page"
  >
    <div
      className="content col-md-8"
    >
      <h1>
        Payment Approved
      </h1>
      <p>
        Thank you for submitting your application to the Criminal Records Review Program.
      </p>
      <p>
        Your application will be reviewed shortly. Once complete, the results will be provided directly to the requesting organization. We will contact you if further information is required.
      </p>
      <div
        className="print"
      >
        <table
          className="white"
          id="print"
        >
          <thead>
            <tr>
              <th
                colSpan="2"
              >
                APPLICATION INFORMATION
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                Service Number
              </td>
              <td>
                987
              </td>
            </tr>
            <tr>
              <td>
                First Name
              </td>
              <td>
                Robert
              </td>
            </tr>
            <tr>
              <td>
                Last Name
              </td>
              <td>
                Ross
              </td>
            </tr>
            <tr>
              <td>
                Organization
              </td>
              <td>
                Public Broadcast Network
              </td>
            </tr>
            <tr>
              <td>
                Amount
              </td>
              <td>
                28.00
              </td>
            </tr>
            <tr>
              <td>
                Date/Time
              </td>
              <td>
                3/4/2020 1:01:36 PM
              </td>
            </tr>
            <tr>
              <td>
                Transaction ID
              </td>
              <td>
                2991
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <button
        className="btn ecrc_go_btn btn btn-sm"
        onClick={[Function]}
        type="submit"
      >
        Print
        <div
          className="loader-hide"
        >
          <div
            className="btn-loader"
          />
        </div>
      </button>
      <button
        className="btn ecrc_go_btn btn btn-sm"
        onClick={[Function]}
        type="submit"
      >
        Download
        <div
          className="loader-hide"
        >
          <div
            className="btn-loader"
          />
        </div>
      </button>
    </div>
    <div
      className="sidecard"
    />
  </div>
  <footer
    className="footer"
  >
    <nav
      className="navbar navbar-expand-sm navbar-dark"
    >
      <div
        className="container"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
              rel="noopener noreferrer"
              target="_blank"
            >
              Disclaimer
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/privacy"
              rel="noopener noreferrer"
              target="_blank"
            >
              Privacy
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/accessibility"
              rel="noopener noreferrer"
              target="_blank"
            >
              Accessibility
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/copyright"
              rel="noopener noreferrer"
              target="_blank"
            >
              Copyright
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/contact-us"
              rel="noopener noreferrer"
              target="_blank"
            >
              Contact Us
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </footer>
</main>
`;

exports[`Success Page Component Matches the Volunteer snapshot 1`] = `
<main>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-none d-md-block"
            height="44"
            src="/images/bcid-logo-rev-en.svg"
            width="181"
          />
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-md-none"
            height="44"
            src="/images/bcid-symbol-rev.svg"
            width="64"
          />
        </a>
        <div>
          <a
            className="navbar-brand nav-item nav-link"
            href="/"
          >
            Criminal Record Check
          </a>
        </div>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNavAltMarkup"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        />
      </div>
    </nav>
  </header>
  <div
    className="page"
  >
    <div
      className="content col-md-8"
    >
      <h1>
        Application Submitted
      </h1>
      <p>
        Thank you for submitting your application to the Criminal Records Review Program.
      </p>
      <p>
        Your application will be reviewed shortly. Once complete, the results will be provided directly to the requesting organization. We will contact you if further information is required.
      </p>
      <div
        className="print"
      >
        <table
          className="white"
          id="print"
        >
          <thead>
            <tr>
              <th
                colSpan="2"
              >
                APPLICATION INFORMATION
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                Service Number
              </td>
              <td>
                987
              </td>
            </tr>
            <tr>
              <td>
                First Name
              </td>
              <td>
                Robert
              </td>
            </tr>
            <tr>
              <td>
                Last Name
              </td>
              <td>
                Ross
              </td>
            </tr>
            <tr>
              <td>
                Organization
              </td>
              <td>
                Public Broadcast Network
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <button
        className="btn ecrc_go_btn btn btn-sm"
        onClick={[Function]}
        type="submit"
      >
        Print
        <div
          className="loader-hide"
        >
          <div
            className="btn-loader"
          />
        </div>
      </button>
      <button
        className="btn ecrc_go_btn btn btn-sm"
        onClick={[Function]}
        type="submit"
      >
        Download
        <div
          className="loader-hide"
        >
          <div
            className="btn-loader"
          />
        </div>
      </button>
    </div>
    <div
      className="sidecard"
    />
  </div>
  <footer
    className="footer"
  >
    <nav
      className="navbar navbar-expand-sm navbar-dark"
    >
      <div
        className="container"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
              rel="noopener noreferrer"
              target="_blank"
            >
              Disclaimer
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/privacy"
              rel="noopener noreferrer"
              target="_blank"
            >
              Privacy
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/accessibility"
              rel="noopener noreferrer"
              target="_blank"
            >
              Accessibility
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/copyright"
              rel="noopener noreferrer"
              target="_blank"
            >
              Copyright
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/contact-us"
              rel="noopener noreferrer"
              target="_blank"
            >
              Contact Us
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </footer>
</main>
`;
